### PR TITLE
Add multiple language stopwords with customizable stop word paths

### PR DESCRIPTION
**Pull request by kreynolds**
This adds the ability to have stop words in multiple languages as well as prepend a custom stopword path. I personally have a much larger stopword list for english that came with this library but I wanted to write everything in a completely backwards compatible way.

### CODE REVIEW

File: lib/classifier-reborn/extensions/hasher.rb [hunk 1]
```diff
@@ -6,8 +6,12 @@
 
 module ClassifierReborn
   module Hasher
+    @stopwords_path = [File.expand_path(File.dirname(__FILE__) + '/../../../data/stopwords')]
```
Comment (parkr) [inline lib/classifier-reborn/extensions/hasher.rb]: make a const?

File: lib/classifier-reborn/extensions/hasher.rb [hunk 2]
```diff
@@ -48,87 +52,18 @@ def word_hash_for_symbols(words)
...
-      within
-      yes
-      you
-      youll
-    ])
+    # Create a lazily-loaded hash of stopword data
+    STOPWORDS = Hash.new do |hash, language|
+      hash[language] = []
+
+      @stopwords_path.each do |path|
```
Comment (parkr) [inline lib/classifier-reborn/extensions/hasher.rb]: should be the method here, no? instead of `@`

File: test/bayes/bayesian_test.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require File.dirname(__FILE__) + '/../test_helper'
 class BayesianTest < Test::Unit::TestCase
 	def setup
-		@classifier = ClassifierReborn::Bayes.new 'Interesting', 'Uninteresting'
+		@classifier = ClassifierReborn::Bayes.new 'Interesting', 'Uninteresting', language: 'en'
```
Comment (parkr) [inline test/bayes/bayesian_test.rb]: this should be the default, right?

**Comment by parkr** [comment]
This is really awesome! How much does this weigh?

Will wait for @Ch4s3's input.

File: test/bayes/bayesian_test.rb (hunk 3 above)
Comment (kreynolds) [inline test/bayes/bayesian_test.rb]: It is the default but I think I wrote that test before there was a default, I'll clean it up.

File: lib/classifier-reborn/extensions/hasher.rb (hunk 1 above)
Comment (kreynolds) [inline lib/classifier-reborn/extensions/hasher.rb]: Sure, consts are fun.

File: lib/classifier-reborn/extensions/hasher.rb (hunk 2 above)
Comment (kreynolds) [inline lib/classifier-reborn/extensions/hasher.rb]: I'll just use the const here, ditch the method

**Comment by kreynolds** [comment]
So when you say 'weigh', you mean does this slow things down any?

**Comment by parkr** [comment]
@kreynolds Yes, but also the byte size increase in downloading the gem.

**Comment by kreynolds** [comment]
Its not any slower, just takes up slightly more memory if you are classifying multiple languages in the same runtime. There are 25K of stopwords among all of the languages put together. Its probably worth noting that after this patch, I have another set of patches to improve performance, particularly around the Hasher (300% speedup, give or take).

File: lib/classifier-reborn/extensions/hasher.rb [hunk 4]
```diff
@@ -18,22 +18,22 @@ def without_punctuation(str)
 
     # Return a Hash of strings => ints. Each word in the string is stemmed,
     # interned, and indexes to its frequency in the document.
-    def word_hash(str)
-      word_hash   = clean_word_hash(str)
+    def word_hash(str, language='en')
```
Comment (parkr) [inline lib/classifier-reborn/extensions/hasher.rb]: Nit: we follow the [GitHub Ruby Styleguide](https://github.com/styleguide/ruby) for new code. It states:

> Use spaces around the = operator when assigning default values to method parameters.

Would you mind updating your changes to match this?

**Comment by kreynolds** [comment]
Done, and I removed Indonesian, which was empty.

**Comment by parkr** [comment]
LGTM. @Ch4s3?

**Comment by Ch4s3** [comment]
Looks great. I'll merge it in as soon as I have time to update the history.

**Comment by parkr** [comment]
:+1:

**Pull request merged**